src/allmydata/scripts/create_node.py

```diff
@@ -344,6 +350,54 @@ def create_node(config):
         os.mkdir(basedir)
     write_tac(basedir, "client")
 
+    # if we're doing magic-wormhole stuff, do it now
+    if config['join'] is not None:
+        print >>out, "Opening wormhole with code '{}'".format(config['join'])
+
+        wh = wormhole.create(
+            appid=u"tahoe-lafs.org/lafs",
+            relay_url=u"ws://wormhole.leastauthority.com:4000/v1",
+            reactor=reactor,
+        )
+        code = unicode(config['join'])
+        wh.set_code(code)
+        yield wh.get_welcome()
+        print >>out, "Connected to wormhole server"
+
+        intro = {
+            u"abilities": {
+                "client-v1": {},
+            }
+        }
+        wh.send_message(json.dumps(intro))
+
+        server_intro = yield wh.get_message()
+        server_intro = json.loads(server_intro)
+
+        print >>out, "  received server introduction"
+        assert u'abilities' in server_intro
+        assert u'server-v1' in server_intro['abilities']
+
+        remote_data = yield wh.get_message()
+        print >>out, "  received configuration"
+
+        # XXX do we want to whitelist what config this can set?
+        remote_config = json.loads(remote_data)
+        print >>out, "Encoding: {needed} of {total} shares, on at least {happy} " + \
+            "servers".format(**remote_config)
+        print >>out, "Overriding the following config:"
+        # presume someone doing "--join" doesn't want to provide
+        # storage; if they do they can change the config after (and/or
+        # allow the 'tahoe invite' to specify storage options?).
+        remote_config['no-storage'] = True
+        remote_config['listen'] = 'none'
+        for k, v in remote_config.items():
+            if k not in ['happy', 'needed', 'total', 'introducer']:
+                print >>out, "   --{}={}".format(k, v)
+            config[k] = v
```

Comment:
This is a bit confusing.  The config doesn't actually map to `tahoe create` command line arguments, right?  This is more of a metaphor to try to explain things to the user?

I don't see an obvious problem with letting the server any supported configuration option... but part of me still wants to err on the side of a whitelist.  Perhaps because I don't have a deep understanding of every single current Tahoe-LAFS option so I'm not confident it's safe to expose them all to the server.  Or perhaps just out of fear of what new behaviors might arise as folks add new configuration options.

Or, from another angle, if the client lets any option be set, the server won't actually know which options it can set on the client.  If a new config option is added to Tahoe-LAFS but `client-v1`  is still in use, a server will have no way to know whether it's talking to a client that supports that option or not.  It might be harmless to set it for everyone or the server might actually care about doing something different for older clients.  So ... I'd go with a whitelist and bump the client version somehow (either the number or add an item to the corresponding object that tells the server what the client's whitelist is).

Also, the handling of `introducer` points in the direction of another reason for a whitelist.  Perhaps a new sensitive value will be added to the configuration at some point.  Without knowing about each key being set, the client has no way to elide that new sensitive value from the logs.